### PR TITLE
samples: add samples for tagging feature

### PR DESCRIPTION
**Pull request by mayurkale22**
Adds a sample for assigning and querying tags. 

Follow-up to https://github.com/googleapis/java-spanner/pull/576

Note: We will add the test when the feature is fully available.

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/spanner/TransactionWithTagSample.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                        .set("FirstName").to("George")
+                        .set("LastName").to("Washington")
+                        .build());
+                return null;
+              }
+            });
+  }
+  // [END spanner_set_transaction_and_request_tags]
+
+  // [START spanner_query_tags]
```
Comment (thiagotnunes) [inline samples/snippets/src/main/java/com/example/spanner/TransactionWithTagSample.java:65]: This one looks like the client spec.

**Review by thiagotnunes — COMMENTED** [review]
It seems that the first sample `spanner_set_transaction_and_request_tags` does not entirely match the one we have in the client spec.

Is that true, or did we decide to change directions @larkee ?

File: samples/snippets/src/main/java/com/example/spanner/TransactionWithTagSample.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  static void queryStats(DatabaseClient databaseClient) {
+    // Execute query on Query statistics
+    // see: https://cloud.google.com/spanner/docs/introspection/query-statistics, for more details.
+    String sql =
+        "SELECT t.REQUEST_TAG, t.AVG_LATENCY_SECONDS, t.AVG_CPU_SECONDS "
```
Comment (olavloite) [inline samples/snippets/src/main/java/com/example/spanner/TransactionWithTagSample.java:70]: As far as I can tell, `REQUEST_TAG` is not yet available in (public) Cloud Spanner databases. This should therefore not be merged until the column is available.

Furthermore: The `t.` prefix here does not work. There's no alias `t` in the query below.

**Comment by mayurkale22** [comment]
> It seems that the first sample `spanner_set_transaction_and_request_tags` does not entirely match the one we have in the client spec.
> 
> Is that true, or did we decide to change directions @larkee ?

Updated the sample wrt the client spec. PTAL

**Comment by thiagotnunes** [comment]
@mayurkale22 could you add some tests to this sample? We would like to make sure everything is working correctly before merging it.

**Comment by mayurkale22** [comment]
> @mayurkale22 could you add some tests to this sample? We would like to make sure everything is working correctly before merging it.

Sure, I will add test soon. I will wait until the new column becomes visible for tests.

**Pull request closed without merging**